boards: add pyboard v1.1 support

kaspar030 opened the pull request:
From the docs:

# Overview    
    
This port adds basic support for the MicroPython pyboard.    
    
Currently the support is pretty minimal, and you might miss many features that    
the original firmware offers. E.g., the board won't show up via USB as either    
serial or mass storage device.    
    
In order to get serial output, connect an USB/serial adapter to USART1 RX/TX    
(Pins X9 / X10).    
    
# How to flash    
    
## put board into DFU mode:    
    
- connect board via USB    
- connect 3.3v to DFU pin (See [official docs](https://github.com/micropython/micropython/wiki/Pyboard-Firmware-Update))    
- press reset button    
    
## Flash as usual:    
    $ BOARD=pyboard make flash"    
    
- unconnect 3.3v/DFU pin    
- reset again    


### Testing procedure

Run the usual tests on a pyboard.

### Issues/PRs references

None.

emmanuelsearch commented:
@kaspar030 nice!! Comments so far:
- I confirm I could build, flash and run
- ran tests/leds but the leds don't blink...
- to term I had to specify PORT=/dev/ttyUSB0

kaspar030 commented:
> * ran tests/leds but the leds don't blink...

Fixed, led was not set up and its macro had the wrong pin number.

> * to term I had to specify PORT=/dev/ttyUSB0

fixed.

kaspar030 commented:
Now also all issues pointed out by Murdock are fixed. @emmanuelsearch could you take another look?

aabadie commented on boards/pyboard/Makefile.features:
```diff
@@ -0,0 +1,13 @@
+# Put defined MCU peripherals here (in alphabetical order)
+#FEATURES_PROVIDED += periph_dma
```
There is no configuration provided for DMA, so I think this should be removed. Same for PWM.

kaspar030 replied:
done

aabadie commented on boards/pyboard/board.c:
```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (C) 2019 Inria
```
Inria has no copyright on this file.

emmanuelsearch replied:
actually Kaspar is doing this work paid by Inria + FU collaboration so should the copyright not be instead Inria and FU ?

kaspar030 replied:
ok, make sense.

aabadie commented on boards/pyboard/include/board.h:
```diff
@@ -0,0 +1,59 @@
+ * @file
+ * @brief       Board specific definitions for the pyboard board
+ *
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
```
Thanks, but you can remove me (and Inria) from the list of authors and copyrights.

emmanuelsearch replied:
(same comment as above)

aabadie commented on boards/pyboard/include/periph_conf.h:
```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (C) 2017 Inria
```
Invalid copyright

emmanuelsearch commented:
works like a charm now!

emmanuelsearch commented:
@kaspar030 can you address the minor comments of @aabadie and squash?

kaspar030 commented:
- adressed comments, squashed

@aabadie I've re-used a lot of the files that you created, thus I kept the authorship. It was impressively easy to get this board working, with all the stm infrastructure in place! But ok, you're out again, so complaints go to me first. :wink:

aabadie commented on boards/pyboard/doc.txt:
```diff
@@ -0,0 +1,31 @@
+- press reset button
+
+## Flash as usual:
+    $ BOARD=pyboard make flash"
```
one last thing: I think this line won't render very well in the generated documentation. There should be a blank line before.
Also note that there's an extra `"` at the end of the line.

kaspar030 replied:
thanks for actually looking at the docs. fixed & amended.

emmanuelsearch commented:
retested, works, let's go.

cladmi commented:
Just found out about the `dfu.inc.mk` while rebasing https://github.com/RIOT-OS/RIOT/pull/8838 it would have been good to split it in another pull request or have an issue for updating the boards already using `dfu-util`.

aabadie commented:
> it would have been good to split it in another pull request or have an issue for updating the boards already using dfu-util

I thought the same when reviewing but I didn't want to add more delay on this when I ACK'ed this PR. I'll update the related boards (nz32-sc151, bluepill, spark-core) and open a PR tomorrow.
